src/common/cluster-store.ts

```diff
@@ -83,21 +81,13 @@ export class ClusterStore extends BaseStore<ClusterStoreModel> {
     }
   }
 
-  handleStateChange = (event: any, clusterId: string, state: ClusterState) => {
-    logger.silly(`[CLUSTER-STORE]: received push-state at ${location.host} (${webFrame.routingId})`, clusterId, state);
-    this.getById(clusterId)?.setState(state);
-  };
-
   registerIpcListener() {
     logger.info(`[CLUSTER-STORE] start to listen (${webFrame.routingId})`);
+    const ipc = ipcMain ?? ipcRenderer;
 
-    if (ipcMain) {
-      this.disposer.push(ipcMainOn(ClusterStore.StateChannel, this.handleStateChange));
-    }
-
-    if (ipcRenderer) {
-      this.disposer.push(ipcRendererOn(ClusterStore.StateChannel, this.handleStateChange));
-    }
+    ipc?.on("cluster:state", (event, clusterId: ClusterId, state: ClusterState) => {
```

Comment:
Why `ipc?.`? It might not exist?

Reply:
It might not exist in testing. Could be wrong though